sycl/plugins/hip/pi_hip.hpp

```diff
@@ -370,18 +372,47 @@ struct _pi_mem {
 ///
 struct _pi_queue {
   using native_type = hipStream_t;
-
-  native_type stream_;
+  static constexpr int default_num_compute_streams = 64;
+  static constexpr int default_num_transfer_streams = 16;
```

Comment:
The number of concurrent compute/DMA streams on AMD devices (gfx906, gfx908, gfx90a, etc, i.e., per device or per GCD in case of multiple per device limits) limits to 4 IIRC. Just to have efficient mapping of the above vectors sizes to hardware limited count(4). 

The choice for the above 64 & 16 were just motivated by the fact of CUDA hardware concurrent streams (Volta = 16, Ampere = 32) and AMD hardware count being 4. Hence reduced by the factor of half, i.e., just a arbitrary and also ensuring `default_num_compute_streams` and `default_num_transfer_streams` were just multiples of 4.

Please let me know if you just prefer sticking with the same count as CUDA or anything else, since the above choice is not really a sound one.

Reply:
I am fine with these numbers.